curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java

```diff
@@ -111,11 +112,26 @@ public void stateChanged(CuratorFramework client, ConnectionState newState)
         {
             if ( newState == ConnectionState.RECONNECTED )
             {
+                if ( debugReconnectLatch != null )
+                {
+                    try
+                    {
+                        debugReconnectLatch.await();
+                    }
+                    catch ( InterruptedException e )
+                    {
+                        Thread.currentThread().interrupt();
+                        e.printStackTrace();
+                    }
+                }
                 createNode();
             }
         }
     };
 
+    @VisibleForTesting
+    volatile CountDownLatch debugReconnectLatch = null;
```

Comment:
Can't predict what thread it will come from. In the current use case it doesn't need to be, but who knows in the future.